src/shared/utils.ts

```diff
@@ -1490,3 +1491,14 @@ export function newVote(voteType: VoteType, myVote?: number): number {
     return myVote == -1 ? 0 : -1;
   }
 }
+
+function sleep<T>(millis: number): Promise<T> {
+  return new Promise(resolve => setTimeout(resolve, millis));
+}
+
+/**
+ * Polls / repeatedly runs a promise, every X milliseconds
+ */
+export function poll(promiseFn: any, millis: number) {
+  promiseFn().then(sleep(millis).then(() => poll(promiseFn, millis)));
```

Comment:
I think this would be easier to read with awaits.

Reply:
K fixed.